### DISCUSSION TITLE
Disable editing but allow highlighting

### DISCUSSION THREAD

**Discussion by Georg7** [ueberdosis/tiptap#3668] [General Questions]
Is there a way disallow all editing but allow highlighting?

**Context:**
Content is shared across users. Users have read-only or edit permissions. However, users with read-only access should still be able to add highlights.

**Comment by aalian66loop** [comment]
Did you find any solution for this

**Reply by SaavanFreed** [reply]
+1 also interested

**Reply by hgrias** [reply]
bump. also interested

**Reply by tomrehnstrom** [reply]
+1

**Comment by SaavanFreed** [comment]
+1 interested

**Comment by tomrehnstrom** [comment]
This is the solution I landed on for this for anyone else stumbling in here.
```tsx
  const editor = useEditor({
    extensions: [StarterKit, Highlight],
    content: 'TEST CONTENT',
    editable: false // Dont allow editing
  })
```

Highlighting selected text still works with this
```tsx
onClick={() => editor.chain().focus()['toggleHighlight']().run()}
```

I also found it nice to have a `BubbleMenu` available for highlighting, but you have to modify it to make it appear when editor is not editable

```tsx
<BubbleMenu
  editor={editor}
  shouldShow={({ state, from, to }) => {
    const { doc, selection } = state
    const { empty } = selection

    // Sometime check for `empty` is not enough.
    // Doubleclick an empty paragraph returns a node size of 2.
    // So we check also for an empty text size.
    const isEmptyTextBlock = !doc.textBetween(from, to).length && isTextSelection(state.selection)

    if (empty || isEmptyTextBlock) {
      return false
    }

    return true
  }}
>
```
This is based on the source implementation for `shouldShow` but with the editable checks removed
